Packs/AzureKeyVault/Integrations/AzureKeyVault/AzureKeyVault.yml

```diff
@@ -0,0 +1,1171 @@
+category: Network Security
+commonfields:
+  id: AzureKeyVault
+  version: -1
+configuration:
+- display: Client ID
+  name: client_id
+  required: true
+  type: 0
+- display: Client Secret (only when using self-deployed Azure application)
+  name: client_secret
+  required: false
+  type: 4
+- display: Use a self-deployed Azure application
+  hidden: false
+  name: self_deployed
+  required: false
+  type: 8
+- display: Tenant ID
+  name: tenant_id
+  required: true
+  type: 0
+- display: Subscription ID
+  name: subscription_id
+  required: true
+  type: 0
+- display: Resource Group Name
+  name: resource_group_name
+  required: true
+  type: 0
+- display: Trust any certificate (not secure)
+  hidden: false
+  name: insecure
+  required: false
+  type: 8
+- display: Use system proxy settings
+  hidden: false
+  name: proxy
+  required: false
+  type: 8
+- display: Fetches credentials
+  hidden: false
+  name: isFetchCredentials
+  required: false
+  type: 8
+- display: Key Vault names - comma seperated list of Key Vaults to fetch secrets from.
+  hidden: false
+  name: key_vaults
+  required: false
+  type: 0
+- display: Secret names - comma seperated list of secrets to fetch.
+  hidden: false
+  name: secrets
+  required: false
+  type: 0
+description: Use Key Vault to safeguard and manage cryptographic keys and secrets
+  used by cloud applications and services.
+display: Azure Key Vault
+name: AzureKeyVault
+script:
+  commands:
+  - arguments:
+    - default: false
+      description: Key Vault name.
+      isArray: false
+      name: vault_name
+      required: true
+      secret: false
+    - auto: PREDEFINED
+      default: false
+      description: Key Vault supported Azure location. The location cannot be changed
+        after the Key Vault is created. Default value is 'westus'.
+      isArray: false
+      name: location
+      predefined:
+      - eastus
```

Comment:
what does the location do? how does it affect the functionality of the vault?

Reply:
The location represents a regional place where the key vault resource resides in. Technically, it will be better to save the Key vault in a closer location to the service that uses the Key Vault (for a better performance), but generally it does affect on the availability of the key vault to another regions.